lib/validation/validator.rb

```diff
@@ -1,132 +1,172 @@
 module Validation
-  module Rules
-    # A hash of rules for this object
-    def rules
-      @rules ||= {}
-    end
+  module Validator
+    attr_reader :object, :results
 
-    # A hash of errors for this object
-    def errors
-      @errors ||= {}
+    def initialize(object)
+      raise ArgumentError, "cannot validate nil" if object.nil?
```

Comment:
This seems strange. Can you explain the reasoning? Shouldn't `nil` just fail appropriate validations?